src/neighbors.cc

```diff
@@ -1367,230 +1351,22 @@ neighborDumpPeers(StoreEntry * sentry)
-
-    if (p->options.weighted_roundrobin)
-        os << " weighted-round-robin";
-
-    if (p->options.mcast_responder)
-        os << " multicast-responder";
-
-#if PEER_MULTICAST_SIBLINGS
-    if (p->options.mcast_siblings)
-        os << " multicast-siblings";
-#endif
-
-    if (p->weight != 1)
-        os << " weight=" << p->weight;
-
-    if (p->options.closest_only)
-        os << " closest-only";
-
-#if USE_HTCP
-    if (p->options.htcp) {
-        os << " htcp";
-        if (p->options.htcp_oldsquid || p->options.htcp_no_clr || p->options.htcp_no_purge_clr || p->options.htcp_only_clr) {
-            bool doneopts = false;
-            if (p->options.htcp_oldsquid) {
-                os << (doneopts ? ',' : '=') << "oldsquid";
-                doneopts = true;
-            }
-            if (p->options.htcp_no_clr) {
-                os << (doneopts ? ',' : '=') << "no-clr";
-                doneopts = true;
-            }
-            if (p->options.htcp_no_purge_clr) {
-                os << (doneopts ? ',' : '=') << "no-purge-clr";
-                doneopts = true;
-            }
-            if (p->options.htcp_only_clr) {
-                os << (doneopts ? ',' : '=') << "only-clr";
-                //doneopts = true; // uncomment if more opts are added
-            }
-        }
-    }
-#endif
-
-    if (p->options.no_netdb_exchange)
-        os << " no-netdb-exchange";
-
-#if USE_DELAY_POOLS
-    if (p->options.no_delay)
-        os << " no-delay";
-#endif
-
-    if (p->login)
-        os << " login=" << p->login;
-
-    if (p->mcast.ttl > 0)
-        os << " ttl=" << p->mcast.ttl;
-
-    if (p->connect_timeout_raw > 0)
-        os << " connect-timeout=" << p->connect_timeout_raw;
-
-    if (p->connect_fail_limit != PEER_TCP_MAGIC_COUNT)
-        os << " connect-fail-limit=" << p->connect_fail_limit;
-
-#if USE_CACHE_DIGESTS
-
-    if (p->digest_url)
-        os << " digest-url=" << p->digest_url;
-
-#endif
-
-    if (p->options.allow_miss)
-        os << " allow-miss";
-
-    if (p->options.no_tproxy)
-        os << " no-tproxy";
-
-    if (p->max_conn > 0)
-        os << " max-conn=" << p->max_conn;
-
-    if (p->standby.limit > 0)
-        os << " standby=" << p->standby.limit;
-
-    if (p->options.originserver)
-        os << " originserver";
-
-    if (p->domain)
-        os << " forceddomain=" << p->domain;
-
-    if (p->connection_auth == 0)
-        os << " connection-auth=off";
-    else if (p->connection_auth == 1)
-        os << " connection-auth=on";
-    else if (p->connection_auth == 2)
-        os << " connection-auth=auto";
-
-    p->secure.dumpCfg(os, "tls-");
-    os << '\n';
-}
-
 static void
 dump_peers(StoreEntry *sentry, CachePeers *peers)
 {
-    char ntoabuf[MAX_IPSTRLEN];
-    int i;
+    PackableStream os(*sentry);
 
     if (!peers) {
-        storeAppendPrintf(sentry, "There are no neighbors installed.\n");
+        os << "There are no neighbors installed.\n";
         return;
     }
```

Comment:
If we wanted to be super strict, the yaml tree should probably be something like:
```
number of neighbours: <number>
neighbours:
  <name1>:
    spec1: value
    spec2: value
    ...
  <name2>:
    spec1: value
    spec2: value
    ...
```

What you propose is good for a human, but might confuse automated clients because there are two different data types used as values for a map.
I don't have any strong opinion, what do you think @yadij ?

Reply:
after 4798ca005c, the output format is now:

```yaml
cache_peers number: 2
cache_peers:
  - name: localhost
    type: sibling
    host: localhost/3129/0
    flags: default tls-disable
    addresses: [ "[::1]" ]
    status: Down
    fetches: 0
    open connections: 0
    average RTT: 0 msec
    pings sent: 0
    pings acked: 0 0%
    replies ignored: 0 0%
    last failed connection at: 13/Apr/2024:04:15:56 +0700
    keep-alive ratio: 0%
  - name: localhost.localdomain
    type: sibling
    host: localhost.localdomain/3130/0
    flags: default tls-disable
    addresses: [  ]
    status: Down
    fetches: 0
    open connections: 0
    average RTT: 0 msec
    pings sent: 0
    pings acked: 0 0%
    replies ignored: 0 0%
    keep-alive ratio: 0%
```
and
```yaml
cache_peers number: 0
```
in case there are no peers configured

Reply:
> the output format is now: ...

Looks good to me overall, thank you! 

I would prefer "cache_peer directives:" to "cache_peers number:", but that is subjective, and I do not insist on that change.

If that does not break YAML, I would probably add an empty line between peers to improve readability, but that is rather subjective and I do not insist on that change.